packages/jest-util/src/fixInstanceOf.ts

```diff
@@ -0,0 +1,11 @@
+export function fixInstanceOfError(error: typeof Error): void {
+  const toString = Object.prototype.toString;
+  const originalHasInstance = error[Symbol.hasInstance];
+  Object.defineProperty(error, Symbol.hasInstance, {
+    value(potentialInstance: any): boolean {
+      return this === error
+        ? toString.call(potentialInstance) === '[object Error]'
+        : originalHasInstance.call(this, potentialInstance);
```

Comment:
note: because jest supports node >= 10 now, I think that `util.types.isNativeError` would work. I did only some limited test and it worked for my small case.